### PR TITLE
Add IOptionsSnapshot example

### PR DESCRIPTION
**Pull request by HaoK**
@divega @glennc 

Fixes: https://github.com/aspnet/Options/issues/156

### CODE REVIEW

**Comment by HaoK** [comment]
@Rick-Anderson FYI

**Comment by Rick-Anderson** [comment]
@divega @glennc can you approve this so I can reference it in https://github.com/aspnet/Docs/pull/2120

File: samples/Config.Options.Snapshot.Web/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+/// <summary>
+/// This sample demonstrates how IOptionsSnapshot will be created per request.
```
Comment (Rick-Anderson) [inline samples/Config.Options.Snapshot.Web/Startup.cs]: /// This sample demonstrates an IOptionsSnapshot being created on the first request
/// after config.json changes. TimeOptions is bound to config.json, but TimeOptions
/// won't be recreated unless the
/// config file changes.  Hit the server and verify the creation time is
/// unchanged between requests.  Modify config.json and the time will
/// be updated on the next request.

**Review by divega — APPROVED** [review]
:shipit:

File: samples/Config.Options.Snapshot.Web/Startup.cs (hunk 1 above)
Comment (divega) [inline samples/Config.Options.Snapshot.Web/Startup.cs]: @HaoK consider adding a readme.md.

**Comment by divega** [comment]
cc @DamianEdwards in case he still wants to reference this.

File: samples/Config.Options.Snapshot.Web/Startup.cs (hunk 1 above)
Comment (HaoK) [inline samples/Config.Options.Snapshot.Web/Startup.cs]: I'll submit this PR and open a new one for adding the readme.md

**Comment by HaoK** [comment]
805e1f308c5f2212a0f83a086c7b94c7766fd228

**Pull request closed without merging**